python/tests/test_mixed_log_reader.py

```diff
@@ -256,16 +251,11 @@ def test_read_overwrite_index(self, data_path):
         MixedLogReader.generate_index_file(str(data_path))
         reader = MixedLogReader(str(data_path), ignore_index=True)
 
-        # By default, we will be generating an index file, overwriting the existing one, so the reader should have
-        # deleted the existing one.
-        assert reader.index is None
-        assert not os.path.exists(reader.index_path)
-
         reader.filter_in_place((PoseMessage,))
         self._check_results(reader, expected_messages)
 
         # Check that we generated a new index.
-        assert os.path.exists(reader.index_path)
+        assert os.path.exists(FileIndex.get_path(data_path))
```

Comment:
Does `index_path` not exist anymore? Any particular reason?

Reply:
It generally doesn't make sense to reference the file directly. It also might be confusing in the case where you specified not to save it, but you did load it.